### ISSUE TITLE
JavaScript error when replacing image in Gallery Widget

### ISSUE THREAD

**Issue by piotrbak** [WordPress/gutenberg#33171]
## Description
When replacing image in Gallery Widget in 5.8-RC1 customizer, there's a JS error happening

## Step-by-step reproduction instructions
1. Go to the customizer
2. Add a gallery widget with a couple of images
3. Publish it
4. Edit one single image and replace it with other from the Media Library
5. See the error

## Expected behaviour
No errors in JS console

## Actual behaviour
The following error appeared:
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in gallery_image_GalleryImage (created by WithDispatch(gallery_image_GalleryImage))
    in WithDispatch(gallery_image_GalleryImage)
    in Unknown (created by WithSelect(WithDispatch(gallery_image_GalleryImage)))
    in WithSelect(WithDispatch(gallery_image_GalleryImage)) (created by Gallery)
    in li (created by Gallery)
```

## Screenshots or screen recording
![image](https://user-images.githubusercontent.com/26767037/124286268-ab1e1080-db46-11eb-91a2-0c7f3b0a4525.png)

## WordPress information
- WordPress version: 5.8-RC1
- Gutenberg version: Not Installed
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

## Device information
- Device: Desktop
- Operating system: Windows 10
- Browser: Chrome 91.0.4472.124

**Comment by talldan** [comment]
> Add a gallery widget with a couple of images

@piotrbak Just want to check, is that with the gallery block? It shouldn't be possible to add the gallery widget.

**Comment by talldan** [comment]
Oh I see now, I was able to trigger it with the gallery block. It's a React warning, but it doesn't cause any user facing bugs. For some reason the warnings are triggered as errors, that may be for extra visibility.

There's an existing issue to fix these #17355, but it's not a high priority task. There are also some open PRs to refactor the gallery block, so they may result in the issue being resolved.

I'll close this in favour of #17355 and leave a comment there about this.
